Engine/source/gui/controls/guiTextCtrl.cpp

```diff
@@ -188,7 +188,7 @@ void GuiTextCtrl::setText(const char *txt)
    if( !mProfile )
       return;
    
-   if (txt)
+   if (txt && txt != mText)
```

Comment:
Indeed, no point doing a string compare if all you're doing is avoiding a string copy since the string comparison will incur more overhead than the copying.